plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevControllerExtensions.kt

```diff
@@ -135,7 +148,57 @@ suspend fun FeatureDevController.onCodeGeneration(
 
         messenger.sendSystemPrompt(tabId = tabId, followUp = getFollowUpOptions(session.sessionState.phase, InsertAction.ALL))
 
+        session.sendMetricDataTelemetry(
+            MetricDataOperationName.END_CODE_GENERATION.toString(),
+            MetricDataResult.SUCCESS.toString()
+        )
         messenger.sendUpdatePlaceholder(tabId = tabId, newPlaceholder = message("amazonqFeatureDev.placeholder.after_code_generation"))
+    } catch (err: FeatureDevException) {
+        when (true) {
+            (err is GuardrailsException) -> {
+                session.sendMetricDataTelemetry(
+                    MetricDataOperationName.END_CODE_GENERATION.toString(),
```

Comment:
Can we declare the `sendMetricDataTelemetry` method to take the enum values, then convert to string within there? It would reduce the amount of repetition in string conversion we're doing here.